dart/lib/src/sentry_exception_factory.dart

```diff
@@ -0,0 +1,32 @@
+import 'protocol/sentry_stack_trace.dart';
+import 'protocol.dart';
+import 'sentry_stack_trace_factory.dart';
+
+/// class to convert Dart Error and exception to SentryException
+class SentryExceptionFactory {
+  final SentryStackTraceFactory _stacktraceFactory;
+
+  SentryExceptionFactory({SentryStackTraceFactory stacktraceFactory})
+      : _stacktraceFactory = stacktraceFactory;
+
+  SentryException getSentryException(
+    dynamic exception, {
+    StackTrace stackTrace,
+    Mechanism mechanism,
+  }) {
+    if (stackTrace == null && exception is Error) {
+      stackTrace = exception.stackTrace;
+    }
```

Comment:
consider using an `else if` here, if `stackTrace` is null, get from `StackTrace.current` maybe?

https://github.com/getsentry/sentry-dart/projects/1#card-48533365

Reply:
✅